### PR TITLE
install: Add flag to just save manifest and not install

### PR DESCRIPTION
**Pull request by daemon1024**
New flag to save the KubeArmor Manifest file for the cluster env without installing
Also fixed panic when Nodes aren't available for environment detection

Related #109

### CODE REVIEW

**Review by nyrahul — CHANGES_REQUESTED** [review]
TOp Speed! Thanks

File: install/install.go [hunk 1]
```diff
@@ -96,79 +136,155 @@ func K8sInstaller(c *k8s.Client, o Options) error {
...
+		}
+		defer func() {
+			if err := f.Close(); err != nil {
+				fmt.Printf("Error closing file: %s\n", err)
+			}
+		}()
+
+		for _, o := range printYAML {
+			if err := writeToYAML(f, o); err != nil {
+				return err
```
[the remark this answers is not in the document]
Comment (daemon1024) [reply install/install.go:278]: If we return the error, we do print it out with return code one
https://pkg.go.dev/github.com/spf13/cobra#CheckErr

File: cmd/install.go [hunk 2]
```diff
@@ -29,4 +29,6 @@ func init() {
 	installCmd.Flags().StringVarP(&installOptions.Namespace, "namespace", "n", "kube-system", "Namespace for resources")
 	installCmd.Flags().StringVarP(&installOptions.KubearmorImage, "image", "i", "kubearmor/kubearmor:stable", "Kubearmor daemonset image to use")
 	installCmd.Flags().StringVarP(&installOptions.Audit, "audit", "a", "", "Kubearmor Audit Posture Context [all,file,network,capabilities]")
+	installCmd.Flags().BoolVar(&installOptions.Save, "save", false, "Save KubeArmor Manifest ")
```
[the remark this answers is not in the document]
Comment (daemon1024) [reply cmd/install.go:32]: So we mandate using filename with `--save` option?

File: install/install.go [hunk 3]
```diff
@@ -35,52 +43,84 @@ func K8sInstaller(c *k8s.Client, o Options) error {
...
+		fmt.Printf("CRD %s ...\n", hspName)
+		if _, err := c.APIextClientset.ApiextensionsV1().CustomResourceDefinitions().Create(context.Background(), &hspCRD, metav1.CreateOptions{}); err != nil {
+			if !apierrors.IsAlreadyExists(err) {
+				return fmt.Errorf("failed to create CRD %s: %+v", hspName, err)
+			}
+			return fmt.Errorf("CRD %s already exists %+v", hspName, err)
 		}
-		fmt.Printf("CRD %s already exists ...\n", hspName)
+	} else {
+		printYAML = append(printYAML, hspCRD)
```
[the remark this answers is not in the document]
Comment (daemon1024) [reply install/install.go:71]: We had them wrapped under `CreateCustomResourceDefinition` but I decoupled them for accomodating print here. Any ideas here?

File: cmd/install.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nyrahul) [reply cmd/install.go:32]: Lets keep this as it is.

**Review by nyrahul — APPROVED** [review]

**Pull request merged**